webapp/src/app/shared/widgets/number-widget/number-widget.component.ts

```diff
@@ -1,14 +1,38 @@
-import {Component, OnInit} from '@angular/core';
+import {Component, OnInit, ViewChild, ElementRef} from '@angular/core';
 import {AbstractWidget} from '../abstract-widget';
 
 @Component({
 	selector: 'app-number-widget',
 	templateUrl: './number-widget.component.html',
 	styleUrls: ['./number-widget.component.scss', '../widget.scss']
 })
-export class NumberWidgetComponent extends AbstractWidget {
-	constructor() {
-		super();
-	}
+export class NumberWidgetComponent extends AbstractWidget implements OnInit {
+	@ViewChild('numberInput', {static: true}) input!: ElementRef;
+
+	private lastValidValue = '0';
+	
+	setSettings(key: string, event: any) {
+		
+
+		let value = event.target.value;
+		
+		if (event.data) {
+			if (event.data.length > value.length) {
```

Comment:
what is that even doing? google says event.data returns a string with the inserted characters. That means it should never be > value.length because that one contains all characters.

Another Point is setting it to the last valid value. I would prefer it to not set anything in this case and just show a warning/error

Reply:
What is that even doing?
if  event.data === null, then the user backspaced.
if value.length === 0 && event.data.length === 0, then the value was reset by the input property because the input was invalid.

I felt that it would be frustrating to accidentally type in something invalid for it to decide to reset itself.